overlord/snapstate/component.go

```diff
@@ -298,8 +300,20 @@ func (c *componentInstallTaskSet) taskSet() *state.TaskSet {
 		tasks = append(tasks, c.maybeDiscardTask)
 	}
 
+	if len(c.beforeLocalSystemModificationsTasks) == 0 {
+		panic("component install task set should have at least one task before local modifications are done")
```

Comment:
nitpick: do we really need this panic? it will panic anyway in the line after this if there is not anything in the slice, maybe just adding a comment saying it should not be empty would be enough.